src/index.js

```diff
@@ -6,17 +6,16 @@ import { browserHistory } from 'react-router';
 import Routes from './routes';
 import thunk from 'redux-thunk';
 import createLogger from 'redux-logger';
+
 import rootReducer from './reducers/rootReducer';
```

Comment:
Adding a bunch of extra complexity because you _might_ need something down the road is a dangerous path. (In programming, we call this principle "[YAGNI](http://martinfowler.com/bliki/Yagni.html)"). `rootReducer` is not the problem here, it's the middleware stuff. Which is probably just failing because it's not fully implemented. My suggestion is to remove it for now and make it it's own feature and PR in isolation when you're ready for it.